roles/db-copy/tasks/active-copy.yml

```diff
@@ -0,0 +1,155 @@
+  become: yes
+  become_user: "{{ oracle_user }}"
+  tags: active-duplicate
+
+- name: Active-copy | Start auxiliary instance
+  shell: |
+    set -o pipefail
+    {{ oracle_home }}/bin/sqlplus -s / as sysdba << EOF
+    startup nomount pfile={{ oracle_home }}/dbs/init{{ oracle_sid }}.ora force
+    alter system register;
+    host sleep 60
+    EOF
+  environment:
+    ORACLE_HOME: "{{ oracle_home }}"
+    ORACLE_SID: "{{ oracle_sid }}"
+    PATH: "{{ oracle_home }}/bin:/usr/local/bin:/bin:/usr/bin:/usr/local/sbin:/usr/sbin"
+  become: yes
+  become_user: "{{ oracle_user }}"
+  tags: active-duplicate
+
+- name: Active-copy | Set sys password for auxiliary instance
+  shell: |
+    set -o pipefail
+    {{ oracle_home }}/bin/orapwd file={{ oracle_home }}/dbs/orapw{{ db_name }} force=y password={{ sys_pass }}
+  environment:
+    ORACLE_HOME: "{{ oracle_home }}"
+    ORACLE_SID: "{{ oracle_sid }}"
+    PATH: "{{ oracle_home }}/bin:/usr/local/bin:/bin:/usr/bin:/usr/local/sbin:/usr/sbin"
+  become: yes
+  become_user: "{{ oracle_user }}"
+  tags: active-duplicate
+
+- name: Active-copy | Duplicate primary database
+  shell: |
+    set -o pipefail
+    {{ oracle_home }}/bin/rman << EOF
+    connect target "sys/{{ sys_pass }}@//{{ lookup('env', 'PRIMARY_IP_ADDR') }}:{{ listener_port | default(1521, true) }}/{{ db_name }}{% if db_domain|default('', true)|length > 0 %}.{{ db_domain }}{% endif %}"
+    connect auxiliary "sys/{{ sys_pass }}@//{{ lookup('env', 'INSTANCE_IP_ADDR') }}:{{ listener_port | default(1521, true) }}/{{ db_name }}{{ standby_suffix }}{% if db_domain|default('', true)|length > 0 %}.{{ db_domain }}{% endif %}"
+    duplicate target database for standby from active database
+    password file
+    spfile
+      set "db_unique_name"="{{ db_name }}{{ standby_suffix }}"
+    section size {{ section_size | default('32G', true) }}
+    using compressed backupset;
+    EOF
+  environment:
+    ORACLE_HOME: "{{ oracle_home }}"
+    ORACLE_SID: "{{ oracle_sid }}"
+    PATH: "{{ oracle_home }}/bin:/usr/local/bin:/bin:/usr/bin:/usr/local/sbin:/usr/sbin"
+  register: db_copy
+  become: yes
+  become_user: "{{ oracle_user }}"
+  tags: active-duplicate
+
+- name: Active-copy | Duplication results
+  debug: var=db_copy.stdout_lines
+  tags: active-duplicate
+
+- name: Active-copy | Add Oracle Restart configuration
+  shell: |
+    set -o pipefail
+    srvctl add db -d {{ db_name }}{{ standby_suffix }} \
+    -oraclehome {{ oracle_home }} {% if db_domain|default('', true)|length > 0 %}-domain {{ db_domain }}{% endif %} \
+    -spfile {{ oracle_home }}/dbs/spfile{{ db_name }}.ora \
+    -pwfile {{ oracle_home }}/dbs/orapw{{ db_name }} \
+    -role PHYSICAL_STANDBY -startoption MOUNT -stopoption IMMEDIATE \
+    -instance {{ oracle_sid }} -dbname {{ db_name }}
+    srvctl start db -d {{ db_name }}{{ standby_suffix }}
```

Comment:
If a "duplicate database" command just completed, won't the standby already be running?

Reply:
duplicate won't add Oracle Restart configuration that's why it is added after duplication. srvctl start db won't start it since it is already up but will change status to running